src/Controller/HomeController.php

```diff
@@ -2,66 +2,266 @@
 
 namespace App\Controller;
 
+use App\Entity\Answer;
+use App\Entity\Question;
+use App\Entity\Quiz;
+use App\Repository\AnswerRepository;
+use App\Repository\QuestionRepository;
 use App\Repository\QuizRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\Tools\Pagination\Paginator;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
 
 #[Route("/app")]
 class HomeController extends AbstractController
 {
-    #[Route('/', name: 'app_quiz')]
-    public function index(QuizRepository $repository): Response
+    public function __construct(
+        protected QuizRepository         $quizRepository,
+        protected QuestionRepository     $questionRepository,
+        protected AnswerRepository       $answerRepository,
+        protected EntityManagerInterface $entityManager,
+        protected Security               $security,
+    )
     {
-        return $this->render('main.html.twig');
     }
 
-    #[Route('/interview', name: 'app_interview')]
-    public function interview(QuizRepository $repository): Response
+    #[Route('/quiz/{quiz}', name: 'app_quiz', defaults: ['quiz' => null])]
+    public function index(Request $request, ?Quiz $quiz): Response
     {
-        return $this->render('interview.html.twig');
+        $questionId = $request->query->getInt('question');
+        $answerId = $request->query->getInt('answer');
+        $previousQuestionId = $request->query->getInt('previousQuestion');
+
+        $previousQuestion = $previousQuestionId
+            ? $this->getQuizQuestionById($quiz, $previousQuestionId)
+            : new Question() /* as null object*/
+        ;
+
+        $quiz = $quiz ?? $this->getFirstQuiz();
+        if (!empty($answerId)) {
```

Comment:
Optional: maybe it's best not to use empty function for value which always should have int type?
My suggestion: if ($answerId !== 0)